subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/ArtifactDependenciesIntegrationTest.groovy

```diff
@@ -350,10 +348,19 @@ dependencies {
     compile "org.gradle.test:lib:1.0@zip"
     compile "org.gradle.test:dist:1.0"
 }
-task test {
-    doLast {
-        assert configurations.compile.files.collect { it.name } == ['lib-1.0.jar', 'lib-1.0-classifier.jar', 'lib-1.0.zip', 'dist-1.0.zip']
-        def artifacts = configurations.compile.resolvedConfiguration.resolvedArtifacts as List
+
+abstract class CheckArtifacts extends DefaultTask {
```

Comment:
Why was this refactored into an entire class? Can this not be accomplished with a `doLast`?

Applies elsewhere. It seems like a lot of these can be replaced with declaring the task's properties outside of the `doLast` and the task action inside the `doLast`

Reply:
This has been suggested in the past as a more proper and idiomatic way to test these.  

Using a task without defining a custom task class might be a little quicker to write and a little more compact to read, but if this is more reflective of how things should be done IRT best practices, I think it makes it a better test.  Also, if we're going to use lazy inputs injected by the runtime like in Sterling's changes, don't we have to do it this way?

Reply:
So I just tried it out and providers are captured, so this is possible with a custom task without writing a type class, like below:

```groovy
tasks.register("test") {
    def filesProvider = provider { configurations.compile.incoming.files }
    def artifactsProvider = provider { configurations.compile.incoming.artifacts }
    
    doLast {
        assert filesProvider.get().collect { it.name } == ['lib-1.0.jar', 'lib-1.0-classifier.jar', 'lib-1.0.zip', 'dist-1.0.zip']
        def artifacts = artifactsProvider.get().artifacts.collect { it.id.name }

        assert artifacts.size() == 4
        assert artifacts[0].name == 'lib'
        assert artifacts[0].type == 'jar'
        assert artifacts[0].extension == 'jar'
        assert artifacts[0].classifier == null
        assert artifacts[1].name == 'lib'
        assert artifacts[1].type == 'jar'
        assert artifacts[1].extension == 'jar'
        assert artifacts[1].classifier == 'classifier'
        assert artifacts[2].name == 'lib'
        assert artifacts[2].type == 'zip'
        assert artifacts[2].extension == 'zip'
        assert artifacts[2].classifier == null
        assert artifacts[3].name == 'dist'
        assert artifacts[3].type == 'zip'
        assert artifacts[3].extension == 'zip'
        assert artifacts[3].classifier == null
    }
}
```

It's good to know.  In this case, however, looking at both side-by-side I think defining a task type is easier to read - it keeps the declaration of what we'll be testing more separate from the assertions.  It's also really not that much longer.  I'm going to leave it.